src/common/simple_spin.cc

```diff
@@ -18,25 +18,42 @@
 #include <stdint.h>
 #include <pthread.h>
 
-void simple_spin_lock(simple_spinlock_t *lock)
+namespace detail {
```

Comment:
i don't think `detail` namespace would help here, but an anonymous namespace does. and would be better if we can rename `delay` to something like `cpu_relax` or just `pause`.